service/notesservice.php

```diff
@@ -196,6 +181,33 @@ public function delete ($id, $userId) {
         $file->delete();
     }
 
+    private function getSafeTitleFromContent($content) {
+        // prepare content: remove markdown characters and empty spaces
+        $content = preg_replace("/^\s*[*+-]\s+/m", "", $content); // list item
+        $content = preg_replace("/^#+\s+(.*?)\s*#*$/m", "$1", $content); // headline
+        $content = preg_replace("/^(?:=*|-*)$/m", "", $content); // separate line for headline
+        $content = preg_replace("/(\*+|_+)(.*?)\\1/m", "$2", $content); // emphasis
+        $content = preg_replace("/^\s+/m", "", $content); // whitespace at beginning
+        $content = preg_replace("/\s+$/m", "", $content); // whitespace at end
+        $content = trim($content);
```

Comment:
Yes, I can do that. I took the separate trimming from the Android app, since we had to remove markups for the whole content (the list of notes shows a small snippet of the note, beside the title).